packages/angular/src/generators/application/lib/normalize-options.ts

```diff
@@ -17,13 +18,15 @@ export function normalizeOptions(
     : names(options.name).fileName;
 
   let e2eProjectName = `${names(options.name).fileName}-e2e`;
-  const appProjectName = appDirectory.replace(new RegExp('/', 'g'), '-');
+  const appProjectName = appDirectory
+    .replace(/[\/\d]/g, '-')
+    .replace(/[-]+/g, '-');
   if (options.e2eTestRunner !== 'cypress') {
     e2eProjectName = `${appProjectName}-e2e`;
   }
 
-  const appProjectRoot = `${appsDir}/${appDirectory}`;
-  const e2eProjectRoot = `${appsDir}/${appDirectory}-e2e`;
+  const appProjectRoot = join(appsDir, appDirectory);
+  const e2eProjectRoot = join(appsDir, `${appDirectory}-e2e`);
```

Comment:
`appsDir` cannot be empty?
I want to right-click a folder and run the application/library schematic with the directory relative to the root
so I've set empty `appsDir` and `libsDir` in the `nx.json` workspaceLayout but it adds a leading `/` because these lines didn't use `path.join`